src/state.py

```diff
@@ -898,9 +898,23 @@ def _parse_script_secrets(charm: ops.CharmBase) -> dict[str, str] | None:
     """
     script_secret_id = typing.cast(str, charm.config.get(SCRIPT_SECRET_ID_CONFIG_NAME, ""))
     script_secret = typing.cast(str, charm.config.get(SCRIPT_SECRET_CONFIG_NAME, ""))
-    logger.info("script secret label: %s, secret: %s", script_secret_id, script_secret)
     if not script_secret_id and not script_secret:
         return None
+    if script_secret_id and script_secret:
+        raise InvalidSecretError(
+            f"Both {SCRIPT_SECRET_CONFIG_NAME} "
+            f"and {SCRIPT_SECRET_ID_CONFIG_NAME} configuration option set. "
+            "Please remove one."
+        )
+    if script_secret_id and not (
+        juju_version := ops.JujuVersion.from_environ()
+    ) >= ops.JujuVersion("3.3"):
+        raise InvalidSecretError(
+            f"Secrets are not supported in Juju version {juju_version}. "
+            "Please consider upgrading the Juju controller to versions "
+            f">= 3.3 or use the {SCRIPT_SECRET_CONFIG_NAME} configuration "
+            "option."
+        )
```

Comment:
Shouldn't the juju version test be for the config option? I doubt the user could set a user secret on a Juju version that doesn't support user secrets 🙂

Reply:
Actually, this is probably good to add as well since the user could be setting something invalid here, perhaps we should add the juju version check on the config option?

Reply:
> Shouldn't the juju version test be for the config option? I doubt the user could set a user secret on a Juju version that doesn't support user secrets 🙂

Would you mind clarifying what you mean by the above please? I don't quite follow what you mean by "juju version test be for the config option"

Reply:
We have the config option in case user secrets are not supported, so we should not allow that to be used if the charm is deployed on a version of juju that supports secrets